src/which.js

```diff
@@ -47,20 +51,16 @@ function _which(options, cmd) {
 
       if (common.platform === 'win') {
         var baseAttempt = attempt;
-        attempt = baseAttempt + '.exe';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.bat';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.cmd';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
+        var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
+        var pathExtArray = splitPath(pathExtEnv);
+        var len = pathExtArray.length;
```

Comment:
Nowadays, I normally don't cache the length unless I'm writing multiple loops of the same array in a performance sensitive section of code. This was originally a simple copy-paste of a 1 1/2 year old PR, where I was still in the habit.

(Cheap checks first are often faster in my experience unless you're dealing with large arrays. But that doesn't apply here.)